agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4326,11 +4326,12 @@ bool slave_thread::autoconfig_wsc_authenticate(std::shared_ptr<ieee1905_1::tlvWs
         return false;
     }
 
-    uint8_t buf[m1_auth_buf_len + m2->getLen() - sizeof(WSC::sWscAttrKeyWrapAuthenticator)];
+    // This is the content of M1 and M2, without the type and length.
+    uint8_t buf[m1_auth_buf_len + m2->getLen() - 3 - sizeof(WSC::sWscAttrKeyWrapAuthenticator)];
     auto next = std::copy_n(m1_auth_buf, m1_auth_buf_len, buf);
     m2->class_swap(); //swap to get network byte order
-    std::copy_n(m2->getStartBuffPtr(), m2->getLen() - sizeof(WSC::sWscAttrKeyWrapAuthenticator),
-                next);
+    std::copy_n(m2->getStartBuffPtr() + 3,
```

Comment:
I prefer having `struct 1905TlvHeader` (simliar to `struct Tlv`) defined in the TLVF non-autogenerated files which will be used instead of hard coding '3' here:
```
struct 1905TlvHeader {
    protected:
        uint8_t tlvType;
        uint16_t tlvLength;

    public:
        uint8_t &type() { return tlvType; }

        Tlv *next() { return (Tlv *)((uint8_t *)this + size()); }

        size_t size() { return sizeof(Tlv) + ntohs(tlvLength); }
    } __attribute__((packed));
```

Then change to:
```
    std::copy_n(m2->getStartBuffPtr() + sizeof(1905TlvHeader),
                m2->getLen() - sizeof(1905TlvHeader) - sizeof(WSC::sWscAttrKeyWrapAuthenticator), next);
```